### ISSUE TITLE
Talks have wrong start and end times: Daylight saving time issue

### ISSUE THREAD

**Issue by Rufus125** [EventFahrplan/EventFahrplan#630]
The Austrian Daylight saving time change is on 31th of March.

The Grazer Linuxtage Event 2024 is on 5th and 6th of April.

The App shows the DST settings of the current day at the event date instead of the correct timezone/DST setting of the day.

E.g.: Keynote starts at 9:00 https://pretalx.linuxtage.at/glt24/schedule/#2024-04-06
In the app, with the "use device time" setting enabled the keynote starts at 08:00 

# Expected behavior
- What should happen?
The talk should show at 09:00

If I change the date on my phone to 18th of April, the app shows the correct time of the talks

**Comment by johnjohndoe** [comment]
Thanks for reporting the case.

As the Grazer Linuxtage will happen after March 31st I suppose the app will be showing the correct time.
Meanwhile, you can toggle the "Use device time zone" preference in the "Settings" screen.

Is that good enough of a workaround for you to close the issue?

**Comment by Rufus125** [comment]
Hi!

As a user I want to be shown the correct time whenever I use the app. I think it is very unintuitive that on one day the times are one hour before and on the next day I see another time in the app.
Also: I might want to plan my visit of the GLT before 31th of March.

And to toggle the setting, I need to be aware that something is wrong in the first place.

So I don't think that this is a good workaround, but I do not expect you to solve it immediately, just wanted to make you aware of this problem.

**Comment by johnjohndoe** [comment]
Thanks for the follow up. Do you have a suggestion on how to fix this? The only thought I have is that the backend system could publish the schedule with the currently valid time zone information for the physical event location. Once the time zone changes the backend can update the schedule including the date/time information.

**Comment by Rufus125** [comment]
I think I found the problem: https://github.com/EventFahrplan/EventFahrplan/blob/2fa4517c2dcf64f32228758bae10a1486dd766cf/commons/src/main/java/info/metadude/android/eventfahrplan/commons/temporal/DateFormatter.kt#L120

`val deviceZoneOffset = OffsetDateTime.now().offset`
now refers to 2025-03-25 which has timezone Europe/Vienna which currently is equal to CET (UTC+1)
however, the conference is on 2025-04-25, which also is in timezone Europe/Vienna, however CEST (UTC+2).

The solution seems to be to take into account the date + time of the starting/ending time of the talks during TZ offset calculation to get the correct times.

**Comment by johnjohndoe** [comment]
Thanks for investigating. Which patch do you suggest? Do you want to try out the code change for GLT 2025?

**Comment by johnjohndoe** [comment]
🔗 https://github.com/linuxtage/EventFahrplan/pull/10

**Comment by Rufus125** [comment]
@johnjohndoe Did you have time to take a look at this PR? I now also created one for this repo. I think it should fix it but I am not familiar with the code base.

**Comment by johnjohndoe** [comment]
@Rufus125 No, I haven't made time to look at this concentrated. Thanks for the PR. I will pick it up - hopefully soon.

**Comment by akarsh** [comment]
Hello @Rufus125

I checked your PR and I have the following suggestion for you to help @johnjohndoe in reviewing the PR.

Please think about the following mentioned in the [contribution guide](https://github.com/EventFahrplan/EventFahrplan/blob/master/CONTRIBUTING.md)
> Write unit tests.

- Please check if your changes affect any existing tests.
    - If there are no tests covering the change. Please check if there is a possibility to write new tests.
    - If there are tests covering the change. Please check if the tests needs to be refactored or not.
- Adding before and after screenshot of how your changes fixes the bug.

**Comment by Rufus125** [comment]
I tested the new version with a test schedule.xml with a different DST TZ (date after 2025-10-26) and it seems like the problem is fixed now! I will keep an eye out for this issue on GLT26 and see if something pops up. Thanks a lot for your dedicated work!

**Comment by johnjohndoe** [comment]
Hm. I don't have time to look into this now. Please report here if you notice something in the future.
